### ISSUE TITLE
utility for video reader and manipulation

### ISSUE THREAD

**Issue by innat** [keras-team/keras#19055]
Unlike [`keras.utils.load_img`](https://github.com/keras-team/keras/blob/v3.0.2/keras/utils/image_utils.py#L187) API for image data reader, currently there is no API for video reader. It would be useful to have this for video modeling dataloader. Currently for video reader in the `tf.data` API, user has to use `tf.py_function` which slow downs the process. Below are some expected utility in such regard

```python
video_file = 'sample.mp4'
num_frames = 20

frames = keras.utils.load_video(video_file) # (100, 224, 224, 3)
sub_frames = keras.utils.uniform_temporal_subsample(
    frames, num_frames, temporal_dim=-4
) # (20, 224, 224, 3)
```

- torch uniform_temporal_subsample [references](https://pytorch.org/vision/stable/generated/torchvision.transforms.v2.UniformTemporalSubsample.html).
- tf uniform_temporal_subsample  [references](https://gist.github.com/innat/205075992360d8d7a241c7f1013866a8)
- video reader: [decord](https://github.com/dmlc/decord), opencv, [av](https://github.com/PyAV-Org/PyAV)

**Comment by innat** [comment]
Yes, I'm willing to contribute. But which tools keras would prefer for video decoding ([decord](https://github.com/dmlc/decord), opencv, [av](https://github.com/PyAV-Org/PyAV))? ( I would recommend decord but it's thrid party. )

**Comment by nkovela1** [comment]
@fchollet Do you have a preference for video decoding? ([decord](https://github.com/dmlc/decord), opencv, [av](https://github.com/PyAV-Org/PyAV))

**Comment by innat-asj** [comment]
@nkovela1 I forgot to mention that, [torch-vision](https://pytorch.org/vision/0.15/io.html) has its own dedicated video utility. But not sure if anything available in tensor-flow or jax. If there are something well performant available, then those can be used instead, leveraging each backend for their own way to decode (similar to onnx support).

Possible video decoder in tensor-flow: [`tfio.experimental.ffmpeg.decode_video`](https://www.tensorflow.org/io/api_docs/python/tfio/experimental/ffmpeg/decode_video). - open issue https://github.com/tensorflow/io/issues/1064

**Comment by Collins-Webdev** [comment]
Bonjour tout le monde 👋
Votre projet me semble bien être très intéressant.
J'aimerais aussi contribuer au projet et plancher notamment sur le problème de utilitaire de lecture et de manipulation vidéo. Veuillez bien vouloir m'ajouter.
Merci d'avance 🤝.

**Comment by innat** [comment]
https://github.com/tensorflow/io/issues/840
